vars/skipStage.groovy

```diff
@@ -232,6 +232,15 @@ boolean call(Map config = [:]) {
     String target_branch = env.CHANGE_TARGET ? env.CHANGE_TARGET : env.BRANCH_NAME
     String tags = config['tags'] ?: parseStageInfo()['test_tag']
 
+    if (config['hw_size']) {
+        // Directly determine if a Functional HW stage should be skipped
+        return skip_ftest_hw(config['hw_size'], target_branch, tags)
+    }
+    if (config['distro']) {
+        // Directly determine if a Functional VM stage should be skipped
+        return skip_ftest(config['distro'], target_branch, tags)
+    }
+
```

Comment:
In the case of functional hardware stages, the only thing that differs from one to the other is the `hw_size`.  So to avoid having to update the `skipStage.groovy` script each time a new stage is needed, why not just directly pass in the `hw_size`.  The functional VM stages are the same, just with the `distro` argument.  While I have not yet used the new `getFunctionalTestStage.groovy` to run the functional VM stages, there is nothing really different other than providing the `distro` input, so its setup if we ever want to use it in the future.

As far as I know, neither `hw_size` nor `distro` are passed in via the `config` argument to `skipStage.groovy`, other than by my changes.  If the concern is the `distro` argument is too general, then we can easily rename it.  Currently `skip_scan_rpms()`, `skip_ftest()`, `skip_ftest_valgrind()`, and `skip_build_bullseye` have a `distro` argument, but they are all called with a hardcoded value.

Reply:
The `hw_size` is the `pragma_suffix` w/o the leading `'-hw``, so each stage currently uses a different `hw_size`.  

This `hw_size` was used in `skip_ftest_hw()` instead of the `pragma_suffix` to easily support checking different commit pragma variations like `Skip-func-test-hw-medium` and `Skip-func-hw-test-medium`, but it could have been done with the `pragma_suffix` and some string manipulation.

Reply:
Ah yes.  I see now.  Thanks for the explanation.